src/test/java/uk/gov/digital/ho/hocs/search/aws/listeners/SearchListenerTest.java

```diff
@@ -0,0 +1,320 @@
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+import uk.gov.digital.ho.hocs.search.api.CaseDataService;
+import uk.gov.digital.ho.hocs.search.api.dto.*;
+import uk.gov.digital.ho.hocs.search.application.aws.SearchListener;
+import uk.gov.digital.ho.hocs.search.application.queue.IndexDataEventRequest;
+import uk.gov.digital.ho.hocs.search.application.queue.EventType;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.HashMap;
+import java.util.UUID;
+
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class SearchListenerTest {
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Mock
+    private CaseDataService caseDataService;
+
+    private final LocalDate now = LocalDate.now();
+
+    @Test
+    public void callsSearchServiceWithValidCreateCaseMessage() throws JsonProcessingException {
+        UUID caseUUID = UUID.randomUUID();
+        CreateCaseRequest createCaseRequest = (new CreateCaseRequest(
+                UUID.randomUUID(),
+                LocalDateTime.now(),
+                "CASE",
+                "CASE/12345",
+                now.plusDays(1),
+                now.minusDays(1),
+                new HashMap()));
+        String data = objectMapper.writeValueAsString(createCaseRequest);
+        IndexDataEventRequest request = new IndexDataEventRequest(caseUUID, data, EventType.CASE_CREATED.text);
+        String message = objectMapper.writeValueAsString(request);
+
+        SearchListener searchListener = new SearchListener(objectMapper, caseDataService);
+        searchListener.onDataEvent(message);
+
+        verify(caseDataService).createCase(eq(caseUUID), eq(createCaseRequest));
+        verifyNoMoreInteractions(caseDataService);
+    }
+
+    @Test
+    public void callsSearchServiceWithValidUpdateCaseMessage() throws JsonProcessingException {
+        UUID caseUUID = UUID.randomUUID();
+        UpdateCaseRequest updateCaseRequest = (new UpdateCaseRequest(
+                UUID.randomUUID(),
+                LocalDateTime.now(),
+                "CASE",
+                "CASE/12345",
+                UUID.randomUUID(),
+                UUID.randomUUID(),
+                now.plusDays(1),
+                now.minusDays(1),
+                new HashMap()));
+        String data = objectMapper.writeValueAsString(updateCaseRequest);
+        IndexDataEventRequest request = new IndexDataEventRequest(caseUUID, data, EventType.CASE_UPDATED.text);
```

Comment:
```suggestion
        IndexDataEventRequest request = new IndexDataEventRequest(caseUUID, data, EventType.CASE_UPDATED.value);
```

Would mean changing the EventType enum, unsure whats more readable?

Reply:
I wasn't sure if value was a protected term - as in there already is one, I'll look at it.